synapse/models/inet.py

```diff
@@ -140,7 +140,7 @@ def _normPyStr(self, valu):
         if len(parts) == 2:
             proto, valu = parts
 
-        if proto not in ('tcp', 'udp', 'icmp', 'host'):
+        if proto not in ('tcp', 'udp', 'icmp', 'host', 'gre'):
```

Comment:
should we define a class specific array of values; that way we can use them in the check here and the error message below?

Reply:
If we're not adding it now we'll still need to update the BadTypeValu message below.